kraken/scenarios/test_pod.py

```diff
@@ -0,0 +1,43 @@
+import logging
+import sys
+import unittest
+
+from kraken.scenarios import kube
+from kraken.scenarios.kube import Client, NotFoundException
+from kraken.scenarios.pod import PodScenario
+
+
+class TestPodScenario(unittest.TestCase):
+    def test_run(self):
+        """
+        This test creates a test pod and then runs the pod scenario restricting the run to that specific pod.
+        """
+        logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+        c = Client()
+        test_pod = c.create_test_pod()
+        self.addCleanup(lambda: self._remove_test_pod(c, test_pod.name, test_pod.namespace))
+
+        scenario = PodScenario(logging.getLogger(__name__))
+        config = scenario.create_config()
+        config.kill = 1
+        config.name_pattern = test_pod.name
+        config.namespace_pattern = test_pod.namespace
+        scenario.run(c, config)
+
+        try:
+            c.get_pod(test_pod.name)
+            self.fail("Getting the pod after a pod scenario run should result in a NotFoundException.")
+        except NotFoundException:
+            return
+
+    @staticmethod
+    def _remove_test_pod(c: kube.Client, pod_name: str, pod_namespace: str):
+        try:
+            c.remove_pod(pod_name, pod_namespace)
+        except NotFoundException:
+            pass
```

Comment:
No other action if trying to remove a non existent pod? Not even logging?

Reply:
No, the intent is for the pod to be gone. It's gone. We can log, but on the other hand, the entire client doesn't log.

Reply:
If I get it right, if NotFoundException is raised here it means that the test pod either didn't got created or crashed while running the scenario. Shouldn't it fail test here in such case?

Reply:
No, `_remove_test_pod` gets called in the cleanup phase, which is there to ensure we don't leave junk in the k8s cluster. See line 19.